src/editor/actions/actions.js

```diff
@@ -160,13 +160,49 @@ export function updateSelectedRevisionId(selectedRevisionId) {
 export function getNotebookRevisionList() {
   return (dispatch, getState) => {
     dispatch({ type: "GETTING_NOTEBOOK_REVISION_LIST" });
+    dispatch({
+      type: "UPDATE_NOTEBOOK_HISTORY_BROWSER_HAS_LOCAL_ONLY_CHANGES",
+      hasLocalOnlyChanges: true
+    });
     getRevisionList(getNotebookID(getState()), isLoggedIn(getState()))
       .then(revisionList => {
-        dispatch({
-          type: "GOT_NOTEBOOK_REVISION_LIST",
-          revisionList
-        });
-        getRequiredRevisionContent(getState(), dispatch);
+        getRevisions(
+          getNotebookID(getState()),
+          [revisionList[0].id],
+          isLoggedIn(getState())
+        )
+          .then(revisions => {
+            if (revisions.length > 0) {
+              if (getState().iomd.trim() === revisions[0].content.trim()) {
+                dispatch({
+                  type: "UPDATE_NOTEBOOK_HISTORY_BROWSER_SELECTED_REVISION_ID",
+                  selectedRevisionId: revisions[0].id
+                });
+                dispatch({
+                  type:
+                    "UPDATE_NOTEBOOK_HISTORY_BROWSER_HAS_LOCAL_ONLY_CHANGES",
+                  hasLocalOnlyChanges: false
+                });
```

Comment:
I don't think this is necessary -- we can calculate it on-demand inside GOT_NOTEBOOK_REVISION_LIST (see my comment inside notebook-reducer.js) as follows:

```js
{
  ...
  revisionList,
  haveLocalOnlyChanges: state.iomd === revisionList[0].content
}
```

In that case, I think we might actually be able to avoid making changes to this file entirely, unless there's something I'm missing.